Library/Homebrew/linkage_checker.rb

```diff
@@ -0,0 +1,121 @@
+require "set"
+require "keg"
+require "formula"
+
+class LinkageChecker
+  attr_reader :keg, :formula
+  attr_reader :brewed_dylibs, :system_dylibs, :broken_dylibs, :variable_dylibs
+  attr_reader :undeclared_deps, :reverse_links
+
+  def initialize(keg, formula = nil)
+    @keg = keg
+    @formula = formula || resolve_formula(keg)
+    @brewed_dylibs = Hash.new { |h, k| h[k] = Set.new }
```

Comment:
These are `dylib`s installed by Homebrew, i.e. from an installed formula. Should be relatively obvious from where this is populated further down. Not sure if there's really a better name.

Do you realize that this and the following comments are about code that isn't actually new and has just moved from one file to another? (I believe you prefer not to modify code while moving.)